File: node/src/age-verification.ts
import { HttpClient } from './http-client';
import type { ShipzyConfig } from './http-client';
import type { ApiResult } from './index';

// ============ Types ============

export type AgeVerificationMethod = 'visual' | 'id_scan_no_copy' | 'digital_id_wallet' | 'other';
export type AgeMinAge = 18 | 21 | 25;

export interface AgeVerificationEvent {
    id: string;
    merchant_id: string;
    parcel_id: string;
    order_id?: string;
    epod_id?: string;
    method: AgeVerificationMethod;
    pass: boolean;
    min_age_required: number;
    checker_user_id: string;
    checked_at: string;
    remark?: string;
    country_code?: string;
    created_at: string;
    updated_at: string;
}

export interface CreateAgeVerificationRequest {
    parcel_id: string;
    order_id?: string;
    epod_id?: string;
    method: AgeVerificationMethod;
    pass: boolean;
    min_age_required: number;
    remark?: string;
    country_code?: string;
}

// ============ Client ============

export class AgeVerificationClient extends HttpClient {
    async create(req: CreateAgeVerificationRequest): Promise<ApiResult<AgeVerificationEvent>> {
        return this.request('/api/v1/age-verifications', 'POST', req);
    }

    async listByParcel(parcelId: string): Promise<ApiResult<{ data: AgeVerificationEvent[] }>> {
        return this.request(`/api/v1/age-verifications?parcel_id=${encodeURIComponent(parcelId)}`);
    }

    async listByOrder(orderId: string): Promise<ApiResult<{ data: AgeVerificationEvent[] }>> {
        return this.request(`/api/v1/age-verifications?order_id=${encodeURIComponent(orderId)}`);
    }
}

File: rust/src/age_verification.rs
use reqwest::Method;

use crate::error::Result;
use crate::http_client::HttpClient;
use crate::types::AgeVerificationEvent;

pub struct AgeVerificationClient {
    pub(crate) inner: HttpClient,
}

impl AgeVerificationClient {
    pub fn new(inner: HttpClient) -> Self {
        Self { inner }
    }
}

impl AgeVerificationClient {
    pub async fn create(
        &self,
        body: serde_json::Value,
    ) -> Result<AgeVerificationEvent> {
        self.inner
            .request("/api/v1/age-verifications", Method::POST, Some(body))
            .await
    }

    pub async fn list_by_parcel(
        &self,
        parcel_id: &str,
    ) -> Result<serde_json::Value> {
        self.inner
            .request(
                &format!(
                    "/api/v1/age-verifications?parcel_id={}",
                    urlencoding::encode(parcel_id)
                ),
                Method::GET,
                None,
            )
            .await
    }

    pub async fn list_by_order(
        &self,
        order_id: &str,
    ) -> Result<serde_json::Value> {
        self.inner
            .request(
                &format!(
                    "/api/v1/age-verifications?order_id={}",
                    urlencoding::encode(order_id)
                ),
                Method::GET,
                None,
            )
            .await
    }
}
